desktop/core/src/desktop/models.py

```diff
@@ -1855,9 +1855,11 @@ def get_main_quick_action(self, apps):
       LOG.exception('Could not load back default app')
 
     if default_interpreter:
-      return default_interpreter[0]
-    else:
-      return default_app
+      if DEFAULT_INTERPRETER.get() != '':
```

Comment:
IIRC no need for `!= ''` check when the config is empty string but good to check.

Reply:
Didn't got you. We want to check when `DEFAULT_INTERPRETER.get()` is not empty right?(`DEFAULT_INTERPRETER.get()` contains the newly set configuration of the default editor)

Reply:
If `DEFAULT_INTERPRETER.get()` is empty it can be considered "falsy" which evaluates to `False` in a boolean context and `True` if it's not empty.